src/components/robot-item-form.tsx

```diff
@@ -0,0 +1,157 @@
+    setFinishLocation('');
+    cleanUpError();
+  };
+
+  const cleanUpError = () => {
+    setNumLoopsError('');
+    setStartLocationError('');
+    setFinishLocationError('');
+  };
+
+  const handleRequestLoop = (event: any) => {
+    event.preventDefault();
+    if (isformValid()) {
+      requestLoop(fleetName, numLoops, startLocation, finishLocation);
+      successMsg('Success');
+      cleanUpForm();
+    }
+  };
+
+  const isformValid = () => {
+    let isValid = true;
+    cleanUpError();
+    if (numLoops === 0 || numLoops < 0) {
+      setNumLoopsError('Loops can only be > 0');
+      isValid = false;
+    }
+    if (!startLocation) {
+      setStartLocationError('Location cannot be empty');
+      isValid = false;
+    }
+    if (!finishLocation) {
+      setFinishLocationError('Location cannot be empty');
+      isValid = false;
+    }
+
+    return isValid;
+  };
+
+  return (
+    <form className={classes.form} onSubmit={handleRequestLoop}>
+      <div className={classes.divForm}>
+        <TextField
+          name="numLoops"
+          onChange={e => e.target.value && setNumLoops(parseInt(e.target.value))}
+          placeholder="Number of loops"
+          type="number"
+          value={numLoops}
+          className={classes.input}
+          label="Number of loops"
+          variant="outlined"
+          error={!!numLoopsError}
+        />
+        {numLoopsError && <p className={classes.error}>{numLoopsError}</p>}
+      </div>
+
+      <div className={classes.divForm}>
+        <Autocomplete
+          getOptionLabel={option => option}
+          onChange={(e, value) => setStartLocation(value || '')}
+          options={listOfPlacesToGo}
+          renderInput={params => (
+            <TextField {...params} label="Pick Start Location" variant="outlined" />
+          )}
+          value={!!startLocation ? startLocation : null}
+        />
+        {startLocationError && <p className={classes.error}>{startLocationError}</p>}
+      </div>
+
+      <div className={classes.divForm}>
+        <Autocomplete
+          getOptionLabel={option => option}
+          onChange={(e, value) => setFinishLocation(value || '')}
+          options={listOfPlacesToGo}
+          renderInput={params => (
+            <TextField {...params} label="Pick Finish Location" variant="outlined" />
+          )}
+          value={!!finishLocation ? finishLocation : null}
+        />
+        {finishLocationError && <p className={classes.error}>{finishLocationError}</p>}
+      </div>
+
+      <div className={classes.buttonContainer}>
+        <Button variant="contained" color="primary" type="submit" className={classes.button}>
+          {'Request'}
+        </Button>
+      </div>
+    </form>
+  );
+};
+
+const useStyles = makeStyles(theme => ({
+  form: {
+    padding: '0.5rem',
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  divForm: {
+    padding: '0.5rem',
+    width: '100%',
+  },
+  error: {
+    color: 'red',
```

Comment:
```suggestion
    color: theme.palette.error.main,
```
We can use the theme color here.

Reply:
Makes sense! Thanks!